src/js/Content/Features/Store/Funds/FCustomGiftcardAndWallet.js

```diff
@@ -30,39 +41,43 @@ export default class FCustomGiftcardAndWallet extends Feature {
         if (giftcard) {
             const styleel = newel.querySelector(".giftcard_style");
             HTML.inner(styleel, Localization.str.wallet.custom_giftcard.replace("__input__", `<span>${input}</span>`));
-            newel.querySelector("#es_custom_money_amount").dataset.tooltipText
+            newel.querySelector("#es_custom_money_input").dataset.tooltipText
                 = Localization.str.wallet.custom_amount_text.replace("__minamount__", price);
 
-            minAmountNode.insertAdjacentElement("afterend", newel);
-
             styleel.addEventListener("click", e => {
-                e.preventDefault();
+                e.stopPropagation();
             });
         } else {
             HTML.inner(priceel, input);
             newel.querySelector("h1").textContent = `${Localization.str.wallet.custom_amount} ${price}`;
             newel.querySelector("p").textContent = Localization.str.wallet.custom_amount_text.replace("__minamount__", price);
-
-            minAmountNode.insertAdjacentElement("afterend", newel);
         }
 
-        newel.querySelector("#es_custom_money_amount").addEventListener("input", e => {
+        minAmountNode.insertAdjacentElement("afterend", newel);
+
+        newel.querySelector("#es_custom_money_input").addEventListener("input", e => {
             let value = e.target.value;
             if (value < minValue) {
                 value = minValue; // prevent purchase error
             }
 
-            const customAmount = Number(value).toFixed(2)
-                .replace(/[,.]/g, "");
+            submitel.dataset.amount = formatValue(value);
 
             if (giftcard) {
                 priceel.textContent = new Price(value);
                 priceel.classList.toggle("small", priceel.textContent.length > 8);
-                newel.querySelector("a.btn_medium").href = `javascript:submitSelectGiftCard( ${customAmount} );`;
             } else {
                 newel.querySelector("h1").textContent = `${Localization.str.wallet.custom_amount} ${new Price(value)}`;
-                newel.querySelector("a.btn_medium").dataset.amount = customAmount;
             }
         });
+
+        // Submit the form to Steam servers, see page scripts `submitSelectGiftCard` or `submitAddFunds`
+        submitel.addEventListener("click", () => {
+            if (!giftcard) {
+                document.querySelector("#input_currency").value = submitel.dataset.currency;
```

Comment:
That's already set correctly on page load? There's also `[name="currency"]` for giftcards

Reply:
Yeah I saw that, but for funds Steam sets it anyway so I copied them.

```js
function submitSelectGiftCard( amount )
{
    $('input_amount').value = amount;
    $('form_addfunds').submit();
}

function submitAddFunds( link )
{
    $('input_amount').value = link.dataset.amount;
    $('input_currency').value = link.dataset.currency;
    $('form_addfunds').submit();
}
````